engine/src/main/java/org/terasology/rendering/dag/NodeFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import org.terasology.context.Context;
+import org.terasology.registry.InjectionHelper;
+
+/**
+ *
+ */
+public class NodeFactory {
+
+    private Context context;
+
+    public NodeFactory(Context context) {
+        this.context = context;
+    }
+
+    public <T extends Node> T createInstance(Class<T> type) {
+        // Attempt constructor-based injection first
+        T node = InjectionHelper.createWithConstructorInjection(type, context);
+        // Then fill @In fields
+        InjectionHelper.inject(node, context);
+        node.initialise();
+        return type.cast(node);
+    }
+}
```

Comment:
The little red thing? I see it on occasion, I suspect it happens when new files are created on different OSes, causing the difference in line/file endings, although Git is supposed to and I guess does handle it for line endings. But maybe it doesn't catch the very final line in some cases.. 

Some user actions can impact it and I think we could explicitly make sure we always have a blank line at the end of everything. I don't think it really causes issues but it is a fair practice to follow, just easy to miss. I'm not sure we have (or can have?) a Checkstyle guard against it, so we often don't notice till the GitHub diff flags it like this.